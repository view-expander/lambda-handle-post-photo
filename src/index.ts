import { APIGatewayProxyResult, S3Event } from 'aws-lambda'
import axios from 'axios'

export async function handler({
  Records,
}: S3Event): Promise<APIGatewayProxyResult> {
  try {
    if (Records.length < 1) {
      throw new Error('No records')
    }

    const { key } = Records[0].s3.object
    const res = await axios
      .post<{ ETag: string }>(
        `${process.env.API_PATH}/skeleton?key=${key}`,
        undefined,
        {
          timeout: 180000,
        }
      )
      .catch((err) => {
        throw err
      })

    return {
      statusCode: res.status,
      body: JSON.stringify(res.data),
    }
  } catch (err) {
    return {
      statusCode: err.response?.status || err.statusCode || 500,
      body: JSON.stringify(err.message),
    }
  }
}
