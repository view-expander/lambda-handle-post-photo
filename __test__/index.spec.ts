import axios from 'axios'
import MockAdapter from 'axios-mock-adapter'
import { handler } from '../src'

describe('handler()', () => {
  let mock: MockAdapter

  beforeEach(() => {
    mock = new MockAdapter(axios)
  })

  afterEach(() => {
    mock.restore()
  })

  it('resoleves dummy response', async () => {
    expect.assertions(1)

    mock.onPost(`${process.env.API_PATH}/skeleton?key=test/key`).reply(200, {
      ETag: '1123456789abcdef0123456789abcdef',
    })

    return expect(
      handler({
        Records: [
          {
            eventVersion: '2.0',
            eventSource: 'aws:s3',
            awsRegion: 'ap-northeast-1',
            eventTime: '1970-01-01T00:00:00.000Z',
            eventName: 'ObjectCreated:Put',
            userIdentity: {
              principalId: 'EXAMPLE',
            },
            requestParameters: {
              sourceIPAddress: '127.0.0.1',
            },
            responseElements: {
              'x-amz-request-id': 'EXAMPLE123456789',
              'x-amz-id-2':
                'EXAMPLE123/5678abcdefghijklambdaisawesome/mnopqrstuvwxyzABCDEFGH',
            },
            s3: {
              s3SchemaVersion: '1.0',
              configurationId: 'testConfigRule',
              bucket: {
                name: 'example-bucket',
                ownerIdentity: {
                  principalId: 'EXAMPLE',
                },
                arn: 'arn:aws:s3:::example-bucket',
              },
              object: {
                key: 'test/key',
                size: 1024,
                eTag: '0123456789abcdef0123456789abcdef',
                sequencer: '0A1B2C3D4E5F678901',
              },
            },
          },
        ],
      })
    ).resolves.toMatchSnapshot()
  })
})
