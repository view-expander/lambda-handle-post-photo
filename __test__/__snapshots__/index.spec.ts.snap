// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handler() resoleves dummy response 1`] = `
Object {
  "body": "{\\"ETag\\":\\"1123456789abcdef0123456789abcdef\\"}",
  "statusCode": 200,
}
`;
